packages/react-components/src/components/Datepicker/helpers.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  isAfter,
+  isSameDay,
+  isSameMonth,
+  subMonths,
+  differenceInCalendarDays,
+} from 'date-fns';
+import { Modifiers } from 'react-day-picker';
+import {
+  IRangeDatePickerProps,
+  IRangeDatePickerState,
+  IRangeDatePickerOption,
+} from './types';
+import cx from 'classnames';
+const baseClass = 'date-picker';
+
+import styles from './DatePicker.module.scss';
+import { ClassNames } from 'react-day-picker/types/ClassNames';
+
+export const isDateWithinRange = (
+  date: Date,
+  range: { from?: Date; to?: Date }
+): boolean => {
+  const { from, to } = range;
+  if (to && !isSameDay(date, to) && isAfter(date, to)) {
+    return false;
+  }
+  if (from && !isSameDay(date, from) && !isAfter(date, from)) {
+    return false;
+  }
+  return true;
+};
+
+export const calculateDatePickerMonth = (
+  initialFromDate?: Date,
+  initialToDate?: Date,
+  toMonth?: Date
+): Date => {
+  if (initialToDate) {
+    const forcePreviousMonth =
+      initialFromDate && !isSameMonth(initialFromDate, initialToDate);
+
+    if (
+      forcePreviousMonth ||
+      (toMonth && isSameMonth(initialToDate, toMonth))
+    ) {
+      return subMonths(initialToDate, 1);
+    }
+
+    return initialToDate;
+  }
+
+  return subMonths(toMonth || new Date(), 1);
+};
+
+export const getRangeDatePickerModifiers = (
+  from?: Date,
+  to?: Date
+): Partial<Modifiers> => {
+  const base = {
+    [styles[`${baseClass}__day--monday`]]: { daysOfWeek: [1] },
+    [styles[`${baseClass}__day--sunday`]]: { daysOfWeek: [0] },
+    [styles[`${baseClass}__day--start`]]: from,
+    [styles[`${baseClass}__day--end`]]: from,
+  };
+
+  if (!to || !from) return base;
+
+  const diff = differenceInCalendarDays(to, from);
+
+  if (diff > 0) {
+    return {
+      ...base,
+      [styles[`${baseClass}__day--end`]]: to,
+    };
+  } else if (diff < 0) {
+    return {
+      ...base,
+      [styles[`${baseClass}__day--start`]]: to,
+    };
+  }
```

Comment:
`else if` is unnecessary here
```suggestion
  if (diff > 0) {
    return {
      ...base,
      [styles[`${baseClass}__day--end`]]: to,
    };
  }
  
  if (diff < 0) {
    return {
      ...base,
      [styles[`${baseClass}__day--start`]]: to,
    };
  }
```

Reply:
👌🏼